spec/classes/ntp_spec.rb

```diff
@@ -3,39 +3,30 @@
 describe 'ntp' do
   let(:facts) {{ :is_virtual => 'false' }}
 
-  ['Debian', 'RedHat', 'Fedora', 'Suse11', 'Suse12', 'FreeBSD', 'Archlinux', 'Gentoo', 'Gentoo (Facter < 1.7)'].each do |system|
-    context "when on system #{system}" do
-      let :facts do
-        case system
-        when 'Gentoo (Facter < 1.7)'
-          super().merge({ :osfamily => 'Linux', :operatingsystem => 'Gentoo' })
-        when 'Suse11'
-          super().merge({ :osfamily => 'Suse', :operatingsystem => 'SLES', :operatingsystemmajrelease => '11' })
-        when 'Suse12'
-          super().merge({ :osfamily => 'Suse', :operatingsystem => 'SLES', :operatingsystemmajrelease => '12' })
-        when 'Fedora'
-          super().merge({ :osfamily => 'RedHat', :operatingsystem => system, :operatingsystemmajrelease => '22' })
-        else
-          super().merge({ :osfamily => system, :operatingsystem => system})
-        end
+  on_supported_os.select { |_, f| f[:os]['family'] != 'Solaris' }.each do |os, f|
+    context "on #{os}" do
+      let(:facts) do
+        f.merge(super())
```

Comment:
Yes, that's intentional. Prior to this, super had defaults that had the os-specific facts added to it. Using rspec-puppet-facts, `f` has ALL facts, and super() becomes more specific, as it is specific for the current test group.